### ISSUE TITLE
Bootstrap in tests/unit not loading

### ISSUE THREAD

**Issue by alairock** [Codeception/Codeception#1223]
_bootstrap.php in tests/unit does not load. Is this the intended behavior? Is there something I need to configure to enable it, or is this a bug?

Codeception 2.0.2

**Comment by maxchirkov** [comment]
I'm running Codeception 2.0.9 and I as well get an Error that a class, which I include via unit/_bootstrap.php, doesn't exist. The test works just fine if I `include(unit/_bootstrap.php)` into the global _bootstrap.php inside `tests` directory. It also runs fine if I run phpunit with bootstrap property of phpunit.xml pointing to unit/_bootstrap.php

**Comment by langj** [comment]
Hi,
I'm running into this error on Codeception 2.0.9.
I debugged it and found the problem in Codecept::runSuite()
https://github.com/Codeception/Codeception/blob/2.0.9/src/Codeception/Codecept.php#L150
``` php
public function runSuite($settings, $suite, $test = null) {
        $suiteManager = new SuiteManager($this->dispatcher, $suite, $settings);
        $suiteManager->initialize();
        $suiteManager->loadTests($test);
        $suiteManager->run($this->runner, $this->result, $this->options);
        return $this->result;
    }
```
The TestFiles are loaded in SuiteManager::loadTests() while the bootstraping is done on the suite.before Event inside of the SuiteManager::run()
(see Codeception\Subscriber\Bootstrap)

As quick fix you can move the event dispatching into SuiteManager::loadTests().
But I dont understand the implications of this, so someone with knowledge should fix it.

**Comment by firejdl** [comment]
I'm still having this issue on Codeception 2.0.14.

**Comment by arall** [comment]
I'm having the same issue on Codeception 2.1.0-beta

**Comment by RenaudParis** [comment]
Still the same issue on Codeception 2.1 stable.

**Comment by RenaudParis** [comment]
Here is my (quick n dirty) workaround, please advise if you have better ideas:

```php
// In /tests/_bootstrap.php
// PATH_TESTS is just "/tests/"
if (!empty($_SERVER['argv'][2]) && $_SERVER['argv'][2] == 'unit')
	require_once(PATH_TESTS . 'unit/_bootstrap.php');
```

**Comment by alexwhitman** [comment]
As others have said this is still occurring in 2.1.  The reason is that `loadBootstrap` at https://github.com/Codeception/Codeception/blob/2.1/src/Codeception/Subscriber/Bootstrap.php#L17 isn't being called for unit tests.  I haven't figured out why yet.

**Comment by manofearth** [comment]
version 2.1.2 same issue

**Comment by manofearth** [comment]
RenaudParis wrote:
> Here is my (quick n dirty) workaround, please advise if you have better ideas:
```php
// In /tests/_bootstrap.php
// PATH_TESTS is just "/tests/"
if (!empty($_SERVER['argv'][2]) && $_SERVER['argv'][2] == 'unit')
    require_once(PATH_TESTS . 'unit/_bootstrap.php');
```

My advise:
```php
if ($_SERVER['argv'][1] === 'run' and in_array('unit', $_SERVER['argv'])) {
	require_once(__DIR__ . '/unit/_bootstrap.php');
}
```

**Comment by Naktibalda** [comment]
I just started a new project using Codeception 2.1.3 and there is no such issue.
Running unit suite always loads `/unit/_bootstrap.php`.

Backtrace:
```
#0  require_once() called at [phar://ProjectDir/codecept.phar/src/Codeception/Subscriber/Bootstrap.php:34]
#1  Codeception\Subscriber\Bootstrap->loadBootstrap()
#2  call_user_func() called at [phar://ProjectDir/codecept.phar/vendor/symfony/event-dispatcher/EventDispatcher.php:164]
#3  Symfony\Component\EventDispatcher\EventDispatcher->doDispatch() called at [phar://ProjectDir/codecept.phar/vendor/symfony/event-dispatcher/EventDispatcher.php:53]
#4  Symfony\Component\EventDispatcher\EventDispatcher->dispatch() called at [phar://ProjectDir/codecept.phar/src/Codeception/SuiteManager.php:152]
#5  Codeception\SuiteManager->run() called at [phar://ProjectDir/codecept.phar/src/Codeception/Codecept.php:203]
#6  Codeception\Codecept->runSuite() called at [phar://ProjectDir/codecept.phar/src/Codeception/Codecept.php:172]
#7  Codeception\Codecept->run() called at [phar://ProjectDir/codecept.phar/src/Codeception/Command/Run.php:262]
#8  Codeception\Command\Run->runSuites() called at [phar://ProjectDir/codecept.phar/src/Codeception/Command/Run.php:189]
#9  Codeception\Command\Run->execute() called at [phar://ProjectDir/codecept.phar/vendor/symfony/console/Command/Command.php:259]
#10 Symfony\Component\Console\Command\Command->run() called at [phar://ProjectDir/codecept.phar/vendor/symfony/console/Application.php:878]
#11 Symfony\Component\Console\Application->doRunCommand() called at [phar://ProjectDir/codecept.phar/vendor/symfony/console/Application.php:195]
#12 Symfony\Component\Console\Application->doRun() called at [phar://ProjectDir/codecept.phar/vendor/symfony/console/Application.php:126]
#13 Symfony\Component\Console\Application->run() called at [phar://ProjectDir/codecept.phar/codecept:34]
#14 require_once(phar://ProjectDir/codecept.phar/codecept) called at [ProjectDir/codecept.phar:7]
```

**Comment by mindplay-dk** [comment]
I'm having this problem too - the `_bootstrap.php` file in `tests/unit` *is* being loaded, but the one in `tests/integration` does *not* get loaded... it's as though it stops trying after finding the boostrap file of one test-suite, the rest don't get loaded? Very odd.

**Comment by DavertMik** [comment]
This should have been fixed somewhere in 2.1.x

**Comment by rybakit** [comment]
Same problem for me. My config files:

codeception.dist.yml:

```yaml
...
settings:
    colors: true
    memory_limit: 1024M
extensions:
...
```
unit.suite.dist.yml:
```yml
...
settings:
    bootstrap: _bootstrap.php
modules:
...
```

I want the `_bootstrap.php` file be loaded only for the unit suite, but it seems to be ignored (Codeception v2.2.9).

**Comment by grahamsutton** [comment]
Also had this problem in Codeception v2.2.9 when performing a CI build with my Laravel app. Only failed for unit tests. Functional tests were fine. Haven't tried with acceptance tests. For me, the problem was a mislabeled folder between a lowercase folder name as compared to a capitalized folder name. See below for what I mean:

Error I got:

```bash
$ ./vendor/bin/codecept run unit -v
Codeception PHP Testing Framework v2.2.9
Powered by PHPUnit 5.7.13 by Sebastian Bergmann and contributors.


  [Codeception\Exception\ConfigurationException]
  Bootstrap file /builds/project-1/tests/unit/_bootstrap.php can't be loaded
```

In my `tests/` directory, I noticed that my folder for unit tests was labeled `myapp/tests/Unit` instead of `myapp/tests/unit` with a lowercase `u`.

Renaming the folder from `tests/Unit/` to `tests/unit` resolved my issue.

I am not sure how isolated this incident is.
